src/clarity-angular/select/providers/select.service.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable, Optional, SkipSelf} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+import {Option} from "./../option";
+
+@Injectable()
+export class RootSelectService {
+    private _changes: Subject<boolean> = new Subject<boolean>();
+    private _input: Subject<string> = new Subject<string>();
+    private _specialKey: Subject<string> = new Subject<string>();
+    private _highlightedChange: Subject<Option> = new Subject<Option>();
```

Comment:
i'm aware of it being a bit heavy and in a earlier approach i was using ids but as options is a changing array these ids weren't usable so the only thing beside passing them i could think of was passing some string unique to that option and then going through all options finding the option matching. I wasn't sure which one would be better speaking of performance myself so i've decided for the cleaner way.  Maybe using a JSON representation of it would be better ?